src/main/java/io/github/debuggyteam/tablesaw/TableSawScreenHandler.java

```diff
@@ -129,4 +132,52 @@ public void close(PlayerEntity player) {
 		this.output.removeStack(0);
 		this.context.run((world, pos) -> this.dropInventory(player, this.input));
 	}
+
+	public void tryCraft(ItemStack stack, boolean multiCraft) {
+		List<TableSawRecipe> recipes = TableSawRecipes.serverInstance().getRecipes(this.input.getStack(0).getItem());
+		for(TableSawRecipe recipe : recipes) {
+			if (ItemStack.areEqual(recipe.getResult(), stack)) {
+				
+				/* This is a complex interlocking series of steps to verify the input count, that the
+				 * output can fit in the destination slot, and then deduct the inputs, and *then* insert the results.
+				 */
+				
+				int availableInputQuantity = this.input.getStack(0).getCount();
+				int availableCraftsFromSource = availableInputQuantity / recipe.getQuantity();
+				if (availableCraftsFromSource<=0) return; //We don't have enough input
```

Comment:
Mostly a nitpick but `availableCraftsFromSource<=0` should be `availableCraftsFromSource <= 0`